io.github.alescdb.mailviewer.json

```diff
@@ -0,0 +1,67 @@
+{
+  "id": "io.github.alescdb.mailviewer",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "47",
+  "sdk": "org.gnome.Sdk",
+  "sdk-extensions": [
+    "org.freedesktop.Sdk.Extension.rust-stable"
+  ],
+  "command": "mailviewer",
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland",
+    "--filesystem=host",
+    "--filesystem=/tmp"
```

Comment:
why does it need `/tmp`?

Reply:
`std::env::temp_dir()` returns `/tmp` (in sandbox). I need `temp_dir()` for extracting attachments when the user clicks on them. If you know a better way, I'm open to suggestions :-)

Reply:
does this need to be shared between different processes outside your app?

Reply:
It depends on what you mean by shared, attachment files are opened with their associated programs using `xdg-open` or  equivalent (crate `open`).

Reply:
the sandbox has its own private `/tmp` that only things within the sandbox can access. So if you put a file there and want a host program to access it, that won't work.

Reply:
So it looks like I need `/tmp`  for now.
I might use `XDG_DOWNLOAD_DIR` in a future version to store attachement files (you can only view them in current version).

Reply:
That is already what I do, I create a `/tmp/.mailviewer.<pid>` when needed and I recursively remove it when the program is close.

Reply:
I've tried to change to `/tmp/mailviewer/<pid>` and flatpak perms to :
- `--filesystem=/tmp/mailviewer`
- `--filesystem=/tmp/mailviewer:create`
- `--filesystem=/tmp/mailviewer:rw`
- `--filesystem=/tmp/mailviewer:rw:create`

but it doesn't seems to work (it works if I create manually `/tmp/mailviewer`)

Reply:
the pid inside the sandbox will always be the same (very likely) and be irrelevant to outside of the sandbox. So there is that.

set env TMPDIR to `/var/tmp` instead.

Reply:
Okay, I'm not sure what I did wrong before, but it seems that `filesystem=/tmp/mailviewer:create` works (I've removed the pid, though I'm concerned that if a user opens more than one instance the files may collide). I’ll try to get rid of tmp in a future version and maybe use `~/.cache/mailviewer` or `XDG_RUNTIME_DIR` instead.

Reply:
> ion and maybe use ~/.cache/mailviewer or XDG_RUNTIME_DIR instead.

You can use that right now by setting TMPDIR=/var/tmp in finish-args. Although that won't be tmpfs so things won't be cleaned up on reboot like your normal host /tmp.

Reply:
>  concerned that if a user opens more than one instance the files may collide)

Add a hash to filename?

Reply:
> Add a hash to filename?

I want to keep the original filename of the attachment, so it looks the same in the title bar of the associate program (loupe, OnlyOffice, ...) thats why I've used pid in the original version. 
Il'l try with an hash of the mail id.

> TMPDIR=/var/tmp

I don't see the difference beetween `/var/tmp` and `/tmp` ?
If I understand you correctly you wan't me to make these changes :

from :
```
    "--filesystem=/tmp/mailviewer:create",
```
with :
```
    "--filesystem=/var/tmp/mailviewer:create",
    "--env=TMPDIR=/var/tmp"
```

Although, now that I've moved from `/tmp/.mailviewer.<pid>` to `/tmp/mailviewer/<pid>` (and now `/tmp/mailviewer`), I have a right issue since I don't have the right to delete `/tmp/mailviewer`, if another user try to launch the program, he won't have the permission to access `/tmp/mailviewer`. 
So I'll need to prefix with user id or something.

Reply:
> with :

No just `--env=TMPDIR=/var/tmp`

`/var/tmp` in the sandbox is an alias to `$XDG_CACHE_HOME/tmp`, so if your implementation follows `TMPDIR`, the file will get automatically created in `$XDG_CACHE_HOME/tmp`. The app already has access to its own `XDG_CACHE_HOME`, no extra permissions are needed.

Reply:
> I want to keep the original filename of the attachment, so it looks the same in the title bar of the associate program (loupe, OnlyOffice, ...) thats why I've used pid in the original version.

this is why in that case you create a temporary directory that remain unique and then put the file in there. See https://crates.io/crates/tempdir

If will repeat what I already said: the PID isn't unique, because it's the PID inside the namespace, and it's very deterministic.

Reply:
> No just `--env=TMPDIR=/var/tmp`

This doesn't work, the actual path is within the sandbox `~/.var/app/<myapp>/cache/tmp`, which external/associated programs cannot access.

Would something like `--filesystem=xdg-run/<myapp>:create` be acceptable to you ?

Reply:
I guess it is fine but also not very different than doing `/tmp/myapp:create`